### ISSUE TITLE
Memory leak when opening IgxGrid pagination dropdown, routing to some other page and coming back to the grid page.

### ISSUE THREAD

**Issue by norikois** [IgniteUI/igniteui-angular#8450]
## Description  
Memory leak when opening IgxGrid pagination dropdown, routing to some other page and coming back to the grid page.

 * igniteui-angular version: 9.1.21, 10.2.0
 * browser: Chrome

## Steps to reproduce  

1. Run the attached sample.
2. Click "grid component" button.
3. Take heap snapshot.
4. Scroll down to the bottom, open the "Items per page" dropdown, and click some other place to close the dropdown.
5. Click "dummy component" button.
6. Repeat the steps from 2 to 5 several times.
7. Compare the number of the Igx components and the number of the HTML elements between each snapshot.

## Result  
The number of the Igx components increases in each snapshot.
The number of the HTML elements increases in each snapshot.

## Expected result  
The numbers do not increase. The "Diff" should be 0.

## Attachments  
[igxGrid-memLeak-test.zip](https://github.com/IgniteUI/igniteui-angular/files/5427957/igxGrid-memLeak-test.zip)
[igxGrid-memLeak-test-Movie.zip](https://github.com/IgniteUI/igniteui-angular/files/5427962/igxGrid-memLeak-test-Movie.zip)

**Comment by wnvko** [comment]
Looks like an issue in `IgxSelect`. We are looking at this now.

**Comment by wnvko** [comment]
@norikois the memory leak is mostly because of the `IgxSelect` which paginator is using. While I was trying to find where the issue is I found that the root of the issue is in @angular/animations. [Here](https://github.com/angular/angular/issues/40191) is the issue I logged to @angular.

What you can do to workaround this issue, until @angular fix it, is next:
- remove the paging of the grid.
- retemplate the paging and hide the dropdown like this
```
<igx-grid #grid [data]="bookListItems" [paging]="true" [perPage]="10" [cellSelection]="'none'"
    [height]="null"  [paginationTemplate]="pager">
    <igx-column [sortable]="true" field="id" header="ID" width="50%">
    </igx-column>
    <igx-column [sortable]="true" field="condition" header="Condition" width="50%">
        <ng-template igxCell let-cell="cell">
            {{bookCondition(cell.value)}}
        </ng-template>
    </igx-column>
</igx-grid>

<ng-template #pager>
    <igx-paginator #paginator [(page)]="grid.page" [totalRecords]="grid.totalRecords" [(perPage)]="grid.perPage"
            [dropdownHidden]="true" [displayDensity]="grid.displayDensity">
    </igx-paginator>
</ng-template>
```

**Comment by Lipata** [comment]
Closing, because either it will be fixed by Angular or #8780
